java/core/src/main/java/com/microsoft/bond/BondType.java

```diff
@@ -245,6 +245,9 @@ protected final void serializeSomethingField(
             StructBondType.StructField<T> field) throws IOException {
         if (value != null) {
             serializeField(context, value.getValue(), field);
+        } else if (!field.isOptional()) {
+            // throws
```

Comment:
Since this is not an explicit "throw", I want to indicate that this line terminates the method.

---
In reply to: [134045874](https://github.com/Microsoft/bond/pull/572#discussion_r134045874) [](ancestors = 134045874)